LibSWBF2/Wrappers/Terrain.cpp

```diff
@@ -187,93 +187,85 @@ namespace LibSWBF2::Wrappers
+					m_Indices.Append(triangleList);
+					vertexOffset += (uint32_t)vertexBuffer->m_TerrainBuffer.Size();
+				}
+				else
 				{
-					// geometry index buffer is optional, most patches don't have them.
-					// this is most likely only for patches with baked in terrain cuts.
-					// if no custom index buffer is specified, we build the index buffer ourselfs
-					IBUF* indexBuffer = p_Terrain->p_Patches->m_Patches[i]->m_GeometryIndexBuffer;
-					VBUF* vertexBuffer = p_Terrain->p_Patches->m_Patches[i]->m_GeometryBuffer;
-					if (indexBuffer != nullptr)
-					{
-						List<uint16_t> triangleList = TriangleStripToTriangleList(indexBuffer->m_IndexBuffer, vertexOffset);
-
-						indices.Append(triangleList);
-						vertexOffset += (uint32_t)vertexBuffer->m_TerrainBuffer.Size();
-					}
-					else
+					// actually z in world space (y is height), but whatever...
+					for (uint16_t y = 0; y < patchEdgeSize; ++y)
 					{
-						// actually z in world space (y is height), but whatever...
-						for (uint16_t y = 0; y < patchEdgeSize; ++y)
+						for (uint16_t x = 0; x < patchEdgeSize; ++x)
 						{
-							for (uint16_t x = 0; x < patchEdgeSize; ++x)
-							{
-								uint32_t globalX = x + vertexOffset;
-								uint32_t globalY = y + vertexOffset;
-
-								uint32_t a, b, c, d;
-
-								// get 4 points for quad
-								a = globalX + (y * dataEdgeSize);
-								b = globalX + ((y + 1) * dataEdgeSize);
-								c = (globalX + 1) + (y * dataEdgeSize);
-								d = (globalX + 1) + ((y + 1) * dataEdgeSize);
-
-								// triangle 1 clockwise
-								indices.Add(a);
-								indices.Add(b);
-								indices.Add(c);
-
-								// triangle 2 clockwise
-								indices.Add(c);
-								indices.Add(b);
-								indices.Add(d);
-							}
+							uint32_t globalX = x + vertexOffset;
+							uint32_t globalY = y + vertexOffset;
+
+							uint32_t a, b, c, d;
+
+							// get 4 points for quad
+							a = globalX + (y * dataEdgeSize);
+							b = globalX + ((y + 1) * dataEdgeSize);
+							c = (globalX + 1) + (y * dataEdgeSize);
+							d = (globalX + 1) + ((y + 1) * dataEdgeSize);
+
+							// triangle 1 clockwise
+							m_Indices.Add(a);
+							m_Indices.Add(b);
+							m_Indices.Add(c);
+
+							// triangle 2 clockwise
+							m_Indices.Add(c);
+							m_Indices.Add(b);
+							m_Indices.Add(d);
 						}
-
-						vertexOffset += verticesPerPatch;
 					}
+
+					vertexOffset += verticesPerPatch;
 				}
 			}
-			else
-			{
-				LOG_WARN("Requested terrain index buffer as (yet) unsupported topology '{}'!", TopologyToString(requestedTopology));
-				return false;
-			}
-
-			lastRequestedTopology = requestedTopology;
+		}
+		else
+		{
+			LOG_WARN("Requested terrain index buffer as (yet) unsupported topology '{}'!", TopologyToString(requestedTopology));
+			return false;
 		}
 
-		count = (uint32_t)indices.Size();
-		indexBuffer = indices.GetArrayPtr();
+		count = (uint32_t)m_Indices.Size();
+		indexBuffer = m_Indices.GetArrayPtr();
 		return true;
```

Comment:
The above changes are mostly git not understanding the de-indent, the only substantive changes were caching, larger indices, and use of the newly generic ```TriangleStripToTriangleList``` helper.